src/FreeformIdeas.js

```diff
@@ -59,19 +65,18 @@ const FreeForm = ({ hideSourceOnDrag }) => {
     const handleClick = useCallback((e) => {
         if(e.target.className !== "FreeformMap") return;
         e.preventDefault();
-        setCoords([coords[0] = e.nativeEvent.layerX, coords[1] = e.nativeEvent.layerY]);
+        setCoords([e.nativeEvent.layerX,e.nativeEvent.layerY]);
         setTimeout(showPopup,300);
-    }, [coords, showPopup]);
+    }, [showPopup]);
 
     return (
-        <div ref={drop} onDoubleClick={handleClick} className='FreeformMap'>
-            <Popup show={popupShow} onClose={hidePopup} onSubmit={addIdea} />
-            <EditPopup   closeEdit={hideEdit} show={editShow} onSubmit={editIdea}/>
+        <div ref={drop} onDoubleClick={handleClick} className="FreeformMap">
+            <Popup show={popupShow} onClose={hidePopup} onSubmit={formType ? addIdea : editIdea} formType={formType}/>
```

Comment:
This works, but you could also just create two `Popup`s.

Reply:
How do you mean? isn't that the same as what I previously did?